vispy/glsl/build_spatial_filters.py

```diff
@@ -277,53 +184,46 @@ class Quadric(SpatialFilter):
 
     Weight function::
 
-             |  0.0 ≤ x < 0.5: 0.75 - x*x
+             |  0 ≤ x < 0.5: 0.75 - x*x
       w(x) = |  0.5 ≤ x < 1.5: 0.5 - (x-1.5)^2
              |  1.5 ≤ x      : 0
 
     """
 
     def __init__(self):
-        SpatialFilter.__init__(self, radius=1.5)
+        super().__init__(radius=1.5)
 
     def weight(self, x):
         if x < 0.75:
             return 0.75 - x * x
         elif x < 1.5:
             t = x - 1.5
             return 0.5 * t * t
-        else:
-            return 0.0
+        return 0
 
 
 class Bicubic(SpatialFilter):
     """
-    Bicubic filter (radius = 2.0).
+    Bicubic filter (radius = 2).
 
     Weight function::
 
       w(x) = 1/6((x+2)^3 - 4*(x+1)^3 + 6*x^3 -4*(x-1)^3)
     """
 
     def __init__(self):
-        SpatialFilter.__init__(self, radius=2.0)
-
-    def pow3(self, x):
```

Comment:
Ah, and it's another set of parantheses missing, the (1/6) is a factor for the whole summation not only the first part.

Reply:
Nice catch, the parenthesis was the issue. I removed the special case for x<0 because `x` can only be positive: `weight` is only called in line 103 (when creating the kernel), and its argument comes from `np.linspace(0, radius, kernel_size), which is always only positive numbers.